code/modules/mob/living/carbon/carbon.dm

```diff
@@ -521,8 +521,10 @@
 	var/stam = getStaminaLoss()
 	if(stam > DAMAGE_PRECISION && (maxHealth - stam) <= crit_threshold && !stat)
 		enter_stamcrit()
+		ADD_TRAIT(src, TRAIT_INCAPACITATED, STAMINA)

```

Comment:
Maybe put the `ADD_TRAIT` where the `stam_paralyzed` variable changes?

Reply:
you're right since `enter_stamcrit` can early return.